### PR TITLE
Finish him.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Resolve the remaining dialyzer errors on repl.  Flawless victory.

### CODE REVIEW

**Comment by seancribbs** [comment]
At first glance these seem ok to me, but I'm not a repl domain expert. This leaves only these remaining errors:

```
riak_core_connection_mgr.erl:494: The pattern 'ok' can never match the type {'error',atom()}
riak_repl_aae_source.erl:574: The pattern <MsgType, Msg, State> can never match since previous clauses completely covered the type <7,binary(),#state{index::'undefined' | non_neg_integer(),indexns::'undefined' | [{_,_}],tree_pid::'undefined' | pid(),tree_mref::'undefined' | reference(),built::'undefined' | non_neg_integer(),timeout::'undefined' | pos_integer(),wire_ver::atom(),diff_batch_size::non_neg_integer(),local_lock::boolean(),owner::'undefined' | pid()}>
riak_repl_console.erl:284: The created fun has no local return
riak_repl_console.erl:285: The pattern {'ok', Members} can never match the type [{string(),non_neg_integer()}]
```

**Comment by cmeiklejohn** [comment]
Those will be addressed in a separate PR.

File: src/riak_repl_keylist_client.erl [hunk 1]
```diff
@@ -257,17 +260,13 @@ send_keylist(continue, #state{kl_fh=FH0,transport=Transport,socket=Socket,kl_cou
...
-    NewState = case Command of
-        cancel_fullsync ->
-            application:unset_env(riak_repl, {progress, SiteName}),
-            State#state{partitions=[], partition=undefined};
-        _ ->
-            State
-    end,
-    log_stop(Command, State),
+wait_ack(cancel_fullsync, #state{sitename=SiteName} = State) ->
+    NewState = application:unset_env(riak_repl, {progress, SiteName}),
```
Comment (Vagabond) [inline src/riak_repl_keylist_client.erl]: This is not the correct value for NewState.

**Comment by Vagabond** [comment]
I don't understand the reason for this ignore:

```
riak_repl_keylist_client.erl:330: The pattern 'cancel_fullsync' can never match the type 'pause_fullsync'
```

**Comment by Vagabond** [comment]
The rest of this looks fine.

**Comment by cmeiklejohn** [comment]
As best as I could tell via Typer, the unset_env and set_env calls which are producing errors because Dialyzer feels they will never succeed, causes it to identify that the `cancel_fullsync` branch will never be reached.

**Comment by Vagabond** [comment]
You could move the log message before the (un)set_env calls, that should shut things up without doing anything other than logging a few instructions earlier.

File: src/riak_repl_keylist_client.erl (hunk 1 above)
Comment (cmeiklejohn) [inline src/riak_repl_keylist_client.erl]: Nice catch.  This was resolved in 6bb6262.

**Comment by cmeiklejohn** [comment]
Addressed logging issues with e30d78f437555e9543bdca8af9d79cca06759150.

**Comment by cmeiklejohn** [comment]
Depends on https://github.com/basho/riak_kv/pull/882.

**Comment by Vagabond** [comment]
+1 on this.

**Pull request merged**